docs/docs/explainers/explainer-writing-noir.md

```diff
@@ -140,6 +140,18 @@ Use arrays and indices that are known at compile time where possible.
 Using `assert_constant(i);` before an index, `i`, is used in an array will give a compile error if `i` is NOT known at compile time.
 :::
 
+### Reduce what is inside loops and conditional logic
+
+Putting less logic  inside an `if` (`else`, etc) paths, or inside a loop, translates to less gates required to represent the program.
+
+A loop duplicates the gates for each iteration of the loop, or put another way, "unwraps" or "flattens" the loop. Any calculations/calls that are unchanged in the loop should be calculated once before, and the result used in the loop.
+
+An `if` will create gates representing each path even though execution will use only one. Furthermore, there are additional operations required for each path. Sometimes this can have a multiplying effect on the operations in the `if` and `else` etc.
```

Comment:
It would be nice to have a small example. Like what does the following translate into? 
```
if cond {
  y = x;
}
```